### PR TITLE
Derive product name from kernel cmdline in init and install scripts

### PR DESCRIPTION
**Pull request by goffioul**
In LIVE mode, the file fstab needs to be pre-processed as the
filesystems are mounted differently. The init script uses hardcoded
filename fstab.androidia_64.

Instead of hardcoded name, derive the filename from
androidboot.hardware.

Jira: https://01.org/jira/browse/AIA-227
Test: Boots to homescreen

Signed-off-by: Michael Goffioul <michael.goffioul@gmail.com>

### CODE REVIEW

**Comment by kalyankondapally** [comment]
@shobhitka ping

**Comment by goffioul** [comment]
Installation does not seem to work (I've been using LIVE boot so far). I may need further changes and adapt this pull request.

**Comment by goffioul** [comment]
Installation now successful.

File: initrd/init [hunk 1]
```diff
@@ -119,6 +119,15 @@ goto_shell()
 		sh 2>&1
 }
 
+product=androidia_64
+for c in `cat /proc/cmdline`; do
+	case $c in
+		androidboot.hardware=*)
+			product=`echo "$c" | cut -b22-`
```
Comment (shobhitka) [inline initrd/init]: All these parameters will come as environment variable I guess and you can use them directly without this complex way of doing it.

Comment (goffioul) [inline initrd/init]: Can you tell me what environment variables you referring to? I've added a "set" call right before dropping into the debug shell, but I don't see any variable related to the kernel cmdline parameters.

Comment (harishkrupo) [inline initrd/init]: The kernel command line parameters are passed down to the init script as environmental variables. Your can access them by using $<variable name>.
For example look at $LIVE, $DEBUG and $INSTALL.

Read this for reference: [kernel command line parameters](https://01.org/linuxgraphics/gfx-docs/drm/admin-guide/kernel-parameters.html)

Comment (goffioul) [inline initrd/init]: The command line parameters are not automatically passed down the init script. It is passed as a variable only if it is not recognized by the kernel and it does not contain '.' character. Hence "androidboot.hardware" is not part of the environment of the init script.

Comment (harishkrupo) [inline initrd/init]: AFAIK androidboot.hardware is not recognized by the kernel, but yeah, need to check that. 👍

Comment (goffioul) [inline initrd/init]: So, are you ok with the method to extract androidboot.hardware from the kernel command line, or do you have another preferred way?

Comment (harishkrupo) [inline initrd/init]: There is a slightly cleaner method but it has a dependency on the version of bash. I believe that the version we are using can do this.
`for i in $(cat /proc/cmdline) ;
do
  case $i in 
    androidboot.hardware=*)
      product="${i//androidboot.hardware=}" ;
  esac ;
done`

Comment (harishkrupo) [inline initrd/init]: This solution is generic and can be used for other variables also. You don't have to count the number of characters to cut :) .

**Comment by goffioul** [comment]
Replaced with #18

**Pull request closed without merging**